src/MICore/TerminalLauncher.cs

```diff
@@ -168,20 +168,36 @@ protected override void SetNewProcessEnvironment(ProcessStartInfo processStartIn
 
     internal class LinuxTerminalLauncher : TerminalLauncher
     {
-        private const string GnomeTerminalPath = "/usr/bin/gnome-terminal";
-        private const string XTermPath = "/usr/bin/xterm";
+        private const string GnomeTerminalPath = "gnome-terminal";
+        private const string XTermPath = "xterm";
         private string _terminalPath;
         private string _bashCommandPrefix;
 
+        private bool ExistInPath(string fileName)
+        {
+            if (File.Exists(fileName))
```

Comment:
This check seems invalid now, unless you are checking /usr/bin/<filename>.

I think the proper thing should be to short circuit it with a /usr/bin/ and if it isn't there, then allow it to search path.

Reply:
You are right, only but it's customary to maintain a backward compatibility 😂